### DISCUSSION TITLE
Limit presale quantity

### DISCUSSION THREAD

**Discussion by fawzydede** [hashlips-lab/nft-erc721-collection#231] [Q&A]
moving [this](https://github.com/hashlips-lab/nft-erc721-collection/issues/142) feature request here.

Can we please have a function to automatically pauses presale after reaching specific amount of supply like `...to.be.revertedWith('Max preSale supply exceeded!');` or something like that to avoid wholesale then not reaching public sale for a small number of collection. cheers!

**Comment by it-is-zods** [comment]
Hashlips has a video that walks you through the contract tweaks needed to have multiple minting phases with different max supplies
https://www.youtube.com/watch?v=fnEsaDUfjpU

**Reply by it-is-zods** [reply]
still early days, but have a look at https://github.com/hashlips-lab/nft-erc721-collection/pull/302

**Reply by it-is-zods** [reply]
It does not pause the contract, but as soon as the minted tokens reach the "minting phase" limit, the smart contract will not allow anyone to mint any further.

The way to do it would be:
1. create your contract
2. call setCurrentMintMaxSupply to initiate the minting in phases and set the new phase limit (*note: this can be done with the contract paused to eliminate the risk of you 
3. run that phase (*note: this can be done using the project's yarn scripts)
4. repeat 2-3 as often as you want
5. if you had enough, you could call setCurrentMintMaxSupply to set the new supply to be the same as the overall contract supply

@liarco, I understand the comment about the UX when there are 1 left and the user tried to mint 3, but this would not be any different for a "minting phase" or when you are reaching the end of the contract / maxLimit.
That said, I might look at the DAPP side of things in the coming days to see how it behaves and if there is any improvement that we can make.

**Reply by fawzydede** [reply]
OMG didn't even remember i made this discussion. Sorry this already solved and yea i made a function setMaxSupply for the workaround. TY

**Reply by it-is-zods** [reply]
can you share the contract id? did it work well with the dapp?
